test/has_shebang.bats

```diff
@@ -11,7 +11,7 @@ setup () {
 @test "has_shebang() - #!/bin/{,a,ba,da,k}sh & #!/bin/bats" {
   source "$PROJECT_ROOT/src/functions.sh"
 
-  local interpreters=$(echo \#\!/bin/{,a,ba,da,k}sh && echo \#\!/bin/bats)
+  local interpreters=( \#\!/bin/{,a,ba,da,k}sh \#\!/bin/bats )
```

Comment:
```suggestion
  local interpreters=( '#!/bin/'{,a,ba,da,k}sh '#!/bin/bats' )
```

Using `'#!'` instead of `\#\!` seems more readable to me:
```console
$ interpreters=( '#!/bin/'{,a,ba,da,k}sh '#!/bin/bats' )
$ echo "${interpreters[@]}"
#!/bin/sh #!/bin/ash #!/bin/bash #!/bin/dash #!/bin/ksh #!/bin/bats
```

Reply:
Yes, I agree, but in some cases it's required to use `\#` and / or `\!` in order to make script run. Since it is shebang sometimes you get:

![Screenshot from 2022-08-03 20-41-15](https://user-images.githubusercontent.com/2879818/182685043-22d83276-ce00-44fb-9aff-7d8ddc3ac53f.png)

Reply:
Updated, looks much better, thanks! :partying_face: